sql database/scripts/createTrain.py

```diff
@@ -11,66 +11,143 @@
 
 # create a connection
 def create_connection(db_file):
-	""" create a db connection to database specified
-		by db_file
-	:param db_file: database file path
-	:return: Connection object or None
-	"""
-	try:
-		conn = sqlite3.connect(db_file);
-		return conn
-	except Error as e:
-		print(e)
-
-	return None
+    """ create a db connection to database specified
+        by db_file
+    :param db_file: database file path
+    :return: Connection object or None
+    """
+    try:
+        conn = sqlite3.connect(db_file);
+        return conn
+    except Error as e:
+        print(e)
+
+    return None
 
 # select all images in the database
 def select_all_images(conn):
-	"""	Query all rows in the images tables
-	:param conn: the Connection object
-	:return: list of rows
-	"""
-	cur = conn.cursor()
-	cur.execute("SELECT Filename FROM images")
-
-	return cur.fetchall()
+    """    Query all rows in the images tables
+    :param conn: the Connection object
+    :return: list of rows
+    """
+    cur = conn.cursor()
+    cur.execute("SELECT Filename FROM images")
+    names = [x[0] for x in cursor.description]
+    return [cur.fetchall(), names]
 
 # select all images with a certain species
 def select_image_by_query(conn, query):
```

Comment:
I think you could improve the function design for selecting the images, since it seems you could just leave "select_all" as the default within the argument definition. 
There doesn't seem to be too much difference between `select_image_by_query` and `select_all_images`, except for the `query` initialization, which could be resolved by having `all` as the default if a query isn't given.